File: mart-stores/src/app/model/product-status.ts
export class ProductStatus {
  public get statusName(): string {
      return this._statusName;
  }
  public set statusName(value: string) {
      this._statusName = value;
  }
  public get id(): number {
      return this._id;
  }
  public set id(value: number) {
      this._id = value;
  }
  constructor(private _id: number, private _statusName: string) {}
}

File: Backend/mart/src/main/java/com/samfrosh/model/ProductStatus.java
package com.samfrosh.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "product_status")
public class ProductStatus {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "status_name")
    private String statusName;

    @OneToMany(cascade = CascadeType.ALL,mappedBy = "productStatus")
    private Set<Product> products;

}